### ISSUE TITLE
AssertionError:Reusage of outer  blob name  y_grad in Do op

### ISSUE THREAD

**Issue by tangdong1994** [caffe2/tutorials#13]
anyone Who has encountered this problem
![wx20181005-195748](https://user-images.githubusercontent.com/37134249/46567278-05bcba80-c963-11e8-8dfa-438cb0fe4d60.png)
![wx20181005-195811](https://user-images.githubusercontent.com/37134249/46567287-1ec56b80-c963-11e8-82bd-bd887c1dd716.png)
![wx20181005-195829](https://user-images.githubusercontent.com/37134249/46567300-559b8180-c963-11e8-96c2-0cd82e3b0212.png)

**Comment by orionr** [comment]
@tangdong1994 do you still see this issue? Thanks.

**Comment by robertparley** [comment]
I also stop at here .
Confusing......

**Comment by orionr** [comment]
@robertparley or @tangdong1994, which tutorial were you running when you saw this? Thanks.

**Comment by robertparley** [comment]
At first, I am try to do some inference with a detectron model. But I need load the pretrain data with C++.
Now I can convert the detectron model to the caffe2 model(init_net.pb & pred.pb)
So what I want to learn exactly is how to load the caffe2 model using C++, and do some predictions.
Oh,It's a heart-broken work for me..........

**Comment by robertparley** [comment]
I read the .prototxt file and found all the types of layers in this networks.
As follow:
Conv
Relu
MaxPool
Sum
ResizeNearest
Sigmoid
GenerateProposals
CollectAndDistributeFpnRpnProposals
RoIAlign
Concat
BatchPermutation
FC
Softmax
BBoxTransform
BoxWithNMSLimit
I don't know whether they are supported by caffe2 now.

**Comment by stackyjoe** [comment]
I'm getting this too. It's the Control_Ops tutorial, second to last runnable cell. I'm on Ubuntu 19.04, I built caffe2 with GPU support. I got the RNN Shakespeare example working(and nvidia-smi confirmed it was using some GPU acceleration), so I think I may have made a mistake during the build process but I'm not entirely sure where.
